### PR TITLE
Fix member assignment in copy constructors

### PR DESCRIPTION
**Pull request by richardbuckle**
Fixes pointed out by Aaron Azhari.

### CODE REVIEW

File: DataDefinitions/DestinationSystem.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class DestinationSystem
 
         public DestinationSystem(DestinationSystem DestinationSystem)
         {
-            this.name = name;
+            this.name = DestinationSystem.name;
```
Comment (Tkael) [inline DataDefinitions/DestinationSystem.cs]: This constructor appears to be entirely unused. Do we need the constructor at all?

If we need it, we should null protect it. 
```cs
             this.name = DestinationSystem?.name; 
```

File: DataDefinitions/Passenger.cs [hunk 2]
```diff
@@ -16,11 +16,11 @@ public class Passenger
...
-            this.missionid = missionid;
-            this.type = type;
-            this.vip = vip;
-            this.wanted = wanted;
-            this.amount = amount;
+            this.missionid = Passenger.missionid;
+            this.type = Passenger.type;
+            this.vip = Passenger.vip;
+            this.wanted = Passenger.wanted;
+            this.amount = Passenger.amount;
```
Comment (Tkael) [inline DataDefinitions/Passenger.cs]: This constructor appears to be entirely unused. Do we need the constructor at all?

If we need it, let's null protect here too.

**Comment by Tkael** [comment]
@Hoodathunk recommends keeping these constructors in place so let's add the null protection then merge this.

**Comment by richardbuckle** [comment]
Microsoft doesn't recommend putting null protection in copy constructors because it requires making properties optional that wouldn't otherwise have to be, and it is often a better design to have non-optional properties. Better to throw the exception and make it the calling code's problem.

**Pull request merged**